spec/src/main/asciidoc/mpconfig.asciidoc

```diff
@@ -0,0 +1,105 @@
+    return ManagedExecutor.builder().maxAsync(a).maxQueued(q).propagated(p).cleared(c).build();
+}
+----
+
+MicroProfile Config can be used to override configuration attributes from the above example as follows,
+
+[source, text]
+----
+exec1.maxAsync=10
+exec1.propagated=Application,CDI
+exec1.cleared=Remaining
+----
+
+=== Container usage of MicroProfile Config
+
+The container produces an instance per unqualified `ManagedExecutor` injection point, which may optionally be annotated with the `@ManagedExecutorConfig` annotation to supply default configuration. The container also produces an instance per `ManagedExecutor` injection point that is annotated with both the `@ManagedExecutorConfig` annotation and the `@NamedInstance` qualifier. And likewise for `ThreadContext` and `@ThreadContextConfig`.
+
+In each of these cases, MicroProfile Config can be used to override the configuration of the instance that is produced by the container.
+
+To override a configuration attribute of an instance that is produced for injection into a field, specify a MicroProfile Config property with the name equal to the fully qualified class name, field name, and configuration attribute name, delimited by the `.` character.
+
+[source, java]
+----
+package org.eclipse.microprofile.example;
+...
+public class MyBean {
+    @Inject
+    ManagedExecutor exec2;
+
+    @Inject @ManagedExecutorConfig(maxAsync=3)
+    ManagedExecutor exec3;
+
+    @Inject @NamedInstance("executor4") @ManagedExecutorConfig(maxAsync=4)
+    ManagedExecutor exec4;
+
+    @Inject @NamedInstance("executor4")
+    ManagedExecutor exec5;
+}
+----
+
+[source, text]
+----
+org.eclipse.microprofile.example.MyBean.exec2.maxAsync=5
+org.eclipse.microprofile.example.MyBean.exec3.maxAsync=6
+org.eclipse.microprofile.example.MyBean.exec4.maxAsync=7
+----
+
+Note that `org.eclipse.microprofile.example.MyBean.exec4.maxAsync` overrides the configuration of the instance that is produced for injection into `exec4`. This same instance is injected into `exec5` per the matching `@NamedInstance("executor4")` qualifier.  It is an error to specify `org.eclipse.microprofile.example.MyBean.exec5.maxAsync`, which will not apply to `exec5`.
+
+To override a configuration attribute of an instance that is produced for injection into a parameter, specify a MicroProfile Config property with the name equal to the fully qualified class name, method name, parameter name, and configuration attribute name, delimited by the `.` character.
+
+[source, java]
+----
+package org.eclipse.microprofile.example;
+...
+public class MyBean {
+    @Produces @ApplicationScoped @NamedInstance("executor6")
+    ManagedExecutor createExecutor(@ManagedExecutorConfig(maxAsync=6) exec) {
+        return exec;
+    }
+
+    @Inject @NamedInstance("executor6")
+    ManagedExecutor exec6;
+}
+----
+
+[source, text]
+----
+org.eclipse.microprofile.example.MyBean.createExecutor.exec.maxAsync=10
```

Comment:
We can't make this property include the parameter name, because method parameter names are only encoded into .class files if some special compiler flag is set.  Otherwise the parameter name at runtime will be `argN` where `N` is the position of the parameter.

I think we have two options here:
1. Use parameter position instead of parameter name, for example: `org.eclipse.microprofile.example.MyBean.createExecutor.0.maxAsync=10`
2. Automatically look for a parameter of type `ManagedExecutor` that is annotated with `@ManagedExecutorConfig`, which would allow us to simply reference the fully-qualified method name, for example:
`org.eclipse.microprofile.example.MyBean.createExecutor.maxAsync=10`
The only limitation with this approach is if a producer wanted to inject two `ManagedExecutor` parameters which were both annotated with `@MEC`. I think this is unlikely enough that it's a reasonable limitation. If someone did have this scenario they could always just look up the MP Config variables themselves.

Reply:
Good catch, I forgot about that detail.  I think with parameter position starting at 1 will make more sense than 0 based on the expectation that the user will count parameters and think in those terms.  Using the method name would be shorter, but also a bit awkward because the method itself doesn't have a maxAsync attribute.